R/kobo.R

```diff
@@ -63,21 +44,115 @@ Kobo <- R6::R6Class("Kobo",
+      } else {
+        usethis::ui_stop(
+          "Invalid version. Must be either v1 or v2.
+          Come back in a couple of years."
+        )
+      }
+    },
+
     #' @description
     #' Example method to send a GET request to the `assets` endpoint
     #' (due to default to `v2`, no further specification is needed).
     get_assets = function() {
-      self$get("assets/")
+      res_list <- self$get("assets/")
+    },
+
+    #' @description
+    #' High-level POST request to clone an asset. `assets` endpoint
+    #' (due to default to `v2`, no further specification is needed).
+    #' @param clone_from character. UID of the asset to be cloned.
+    #' @param name character. Name of the new asset.
+    #' @param asset_type character. Type of the new asset. Can be
+    #' "block", "question", "survey", "template".
+    clone_asset = function(clone_from, name, asset_type) {
+      body = list(
+        "clone_from" = clone_from,
+        "name" = name,
+        "asset_type" = asset_type)
+      self$post("assets/", body = body)
+    },
+
+    #' @description
+    #' High-level POST request to deploy an asset.
+    #' `assets/{uid}/deployment/` endpoint (due to
+    #' default to `v2`, no further specification is needed).
+    #' @param uid character. UID of the asset to be deployed.
+    deploy_asset = function(uid) {
+      body = list("active" = "true")
+      endpoint = paste0("assets/",uid,"/deployment/")
+      self$post(endpoint, body = body)
+    },
+
+    #' @description
+    #' High-level POST request to create an empty asset. `assets/` endpoint
+    #' (due to default to `v2`, no further specification is needed).
+    #' @param name character. Name of the new asset.
+    #' @param description character. Optional.
+    #' @param sector character. Optional.
+    #' @param country character. Optional.
+    #' @param share_metadata boolean. Optional.
+    #' @param asset_type character. Type of the new asset. Can be
+    #' "block", "question", "survey", "template".
+    create_asset = function(name,description,sector,
+                            country,share_metadata,asset_type) {
+      settings = list_as_json_char(
+        list(
+          "description"=description,
+          "sector"=sector,"country"=country,
+           "share_metadata"=share_metadata)
+           )
+      body = list(
+        "name" = name,
+        "settings"=settings,
+        "asset_type" = asset_type)
+      self$post("assets/", body = body)
+    },
+
+    #' @description
+    #' High-level POST request to import an XLS form. `imports` endpoint
+    #' (due to default to `v2`, no further specification is needed).
+    #' @param name character. Name of the new asset.
+    #' @param file character. The path to the file containing the XLS form.
+    import_xls_form = function(name,file) {
+      body = list(
+        "name"=name,
+        "library"="false",
+        "file"=crul::upload(file))
+      self$post("imports/", body = body)
     }
```

Comment:
Maybe `file_path` is a clearer argument name?

```suggestion
    #' @param file_path  character. The path to the XLS form file.
    import_xls_form = function(name, file_path) {
      body = list(
        "name"=name,
        "library"="false",
        "file"=crul::upload(file_path))
      self$post("imports/", body = body)
    }
```